[CALCITE-6883] Move RelRoot tests to dedicated test class

nielspardon opened the pull request:
Follow-up to https://github.com/apache/calcite/pull/4230

zabetak commented on core/src/test/java/org/apache/calcite/rel/RelRootTeast.java:
```diff
@@ -0,0 +1,122 @@
+/**
+ * Tests for {@link RelRoot}.
+ */
+public class RelRootTeast {
```
RelRootTeast -> RelRootTest

nielspardon replied:
oops, should be fixed now

julianhyde commented on core/src/test/java/org/apache/calcite/rel/RelRootTest.java:
```diff
@@ -0,0 +1,122 @@
+    assertFalse(root.isNameTrivial());
+
+    final RelNode project = root.project();
+    assertEquals(inputRel, project);
```
We tend to use `assertThat`. 

Can you remove uses of `assertEquals`, `assertTrue`, `assertFalse`, `assertInstanceOf`, and convert `assertTrue(a)` &rarr; `assertThat(a, is(true))`, `assertEquals(x, a)` &rarr; `assertThat(a, is(x))`, etc.

nielspardon replied:
sure. changed it to assertThat

julianhyde reviewed (commented):
`RelRoot.isNameTrivial()` was unused until your change. Now it is used it needs javadoc, with some useful examples. Can you also add javadoc for `isRefTrivial`, `isCollationTrivial`.

nielspardon commented:
> `RelRoot.isNameTrivial()` was unused until your change. Now it is used it needs javadoc, with some useful examples. Can you also add javadoc for `isRefTrivial`, `isCollationTrivial`.

Done.

nielspardon commented:
> > `RelRoot.isNameTrivial()` was unused until your change. Now it is used it needs javadoc, with some useful examples. Can you also add javadoc for `isRefTrivial`, `isCollationTrivial`.
> 
> Done, I added javadocs to the methods. I added examples for `isNameTrivial()` and `isRefTrivial()`. For `isCollationTrivial()` I couldn't think of a good example. Let me know if you have one I should add.

@julianhyde I haven't heard back from you in a month. anything else I can do on this PR?

zabetak commented:
As far as I see the review comments were addressed and the tests pass so I will merge this PR now.
